public/components/bulkReindexes.react.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { Panel, Button, Table } from 'react-bootstrap';
+import RunningReindex from './runningReindex.react';
+
+
+export default class BulkReindex extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.cancelReindex = this.cancelReindex.bind(this);
+        this.renderCancelButton = this.renderCancelButton.bind(this);
+    }
+
+
+    cancelReindex(reindex, isRunningJob) {
+        this.props.onCancelReindex(reindex, isRunningJob);
+    }
+
+    renderCancelButton(reindex, isRunningJob) {
+        if (isRunningJob && !reindex.settings.supportsCancelReindex) {
+            return (null);
+        } else {
+            return (
+                <Button bsStyle="danger" className="pull-right" type="button" onClick={this.cancelReindex.bind(this, reindex, isRunningJob)}>Cancel</Button>
+            );
+        }
+    }
+
+    render () {
+
+        const completedJobsNodes = this.props.completedReindexes.map(completedJob => {
+            return (
+                <tr key={completedJob.startTime}>
+                    <td>{completedJob.name}</td>
+                    <td>{completedJob.env}</td>
+                    <td>{completedJob.status}</td>
+                    <td>{new Date(completedJob.startTime).toUTCString()}</td>
+                    <td>{new Date(completedJob.finishTime).toUTCString()}</td>
+                </tr>
+            );
+        });
+
+        const runningJobsNodes = this.props.runningReindexes.map(runningJob => {
+            return (
+                <tr key={runningJob.startTime}>
+                    <td>{runningJob.name}</td>
+                    <td>{runningJob.env}</td>
+                    <td>
+                        <RunningReindex documentsIndexes={runningJob.documentsIndexed}
+                                        documentsExpected={runningJob.documentsExpected}
+                                        environment={runningJob.env}
+                                        id={runningJob.id}
+                                        startTime={runningJob.startTime}
+                                        reindex={runningJob}
+                                        isCancelSupported={false}
+                                        onCancelReindex={this.cancelReindex.bind(this)}
```

Comment:
If `onCancelReindex` will never get called, I guess you can pass a noop, like `onCancelReindex={() => {}}`